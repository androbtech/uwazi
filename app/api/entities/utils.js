export function updateMetadataNames(documents, nameMatches) {
  documents.forEach((doc) => {
    const oldMetadata = Object.assign({}, doc.metadata);
    Object.keys(doc.metadata).forEach((propertyName) => {
      if (nameMatches[propertyName]) {
        delete doc.metadata[propertyName];
        doc.metadata[nameMatches[propertyName]] = oldMetadata[propertyName];
      }
    });
  });

  return documents;
}

export function deleteMetadataProperties(documents, properties = []) {
  documents.forEach((doc) => {
    properties.forEach((property) => {
      delete doc.metadata[property];
    });
  });

  return documents;
}

export function removeEntityFilenames(entities) {
  return entities.map((entity) => {
    const safeEntity = { ...entity, file: { ...entity.file } };
    if (entity.file) {
      delete safeEntity.file.filename;
    }
    if (safeEntity.attachments) {
      safeEntity.attachments = safeEntity.attachments.map((attachment) => {
        const safeAttachment = { ...attachment };
        delete safeAttachment.filename;
        return safeAttachment;
      });
    }
    return safeEntity;
  });
}
