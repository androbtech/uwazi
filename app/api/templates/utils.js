import uuid from 'node-uuid';

const generateName = (property) => {
  const name = property.label ? property.label.trim().replace(/[^a-z0-9]/gi, '_').toLowerCase() : property.name;
  return `${name}_${property.type}`;
};

export function generateNames(properties) {
  return properties.map(property => ({
    ...property,
    name: generateName(property)
  }));
}

export function generateIds(properties = []) {
  return properties.map(property => ({
    ...property,
    id: property.id || uuid.v4()
  }));
}

export function generateNamesAndIds(_properties = []) {
  const properties = generateNames(_properties);
  return generateIds(properties);
}

const flattenProperties = properties => properties.reduce((flatProps, p) => {
  if (p.values) {
    return flatProps.concat(p.values);
  }
  return flatProps.concat(p);
}, []);

export function getUpdatedNames(oldProperties = [], newProperties, prop = 'name') {
  const propertiesWithNewName = {};

  flattenProperties(oldProperties).forEach((property) => {
    const newProperty = flattenProperties(newProperties)
    .find(p => p.id === property.id);

    if (newProperty && newProperty[prop] !== property[prop]) {
      propertiesWithNewName[property[prop]] = newProperty[prop];
    }
  });

  return propertiesWithNewName;
}

const includedIn = propertyCollection => property => !propertyCollection.find(p => p.id === property.id);

export function getDeletedProperties(oldProperties = [], newProperties, prop = 'name') {
  return flattenProperties(oldProperties)
  .filter(includedIn(flattenProperties(newProperties)))
  .map(property => property[prop]);
}
