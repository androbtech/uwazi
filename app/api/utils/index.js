// export { default as createError } from './Error';
// export { default as validateRequest } from './validateRequest';
import validateRequest from './validateRequest';
import createError from './Error';
import handleError from './handleError';

export {
  createError,
  validateRequest,
  handleError,
};

export default {
  createError,
  validateRequest,
  handleError,
};
