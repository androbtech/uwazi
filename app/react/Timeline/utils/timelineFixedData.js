const caseTemplate = '58b2f3a35d59f31e1345b48a';
const matterTemplate = '58b2f3a35d59f31e1345b4a4';

const renderableTemplates = {
  judgement: '58b2f3a35d59f31e1345b4ac',
  admissibilityReport: '58b2f3a35d59f31e1345b479',
  orderOfThePresident: '58b2f3a35d59f31e1345b482',
  orderOfTheCourt: '58b2f3a35d59f31e1345b471',
  letterOfSubmission: '59ee461f40f4a54920bd9b87',
  meritsReport: '59ee427f40f4a54920bd9b67'
};

export {
  caseTemplate,
  matterTemplate,
  renderableTemplates,
};
