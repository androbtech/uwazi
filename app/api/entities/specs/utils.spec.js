import { removeEntityFilenames } from '../utils';

describe('entities utils', () => {
  describe('removeEntityFilenames', () => {
    it('should remove filename from main file and attachments for entity', () => {
      const entities = [
        {
          sharedId: 'ent1',
          file: { filename: 'main.pdf', originalname: 'Main.pdf' },
          attachments: [
            { _id: 'att1', filename: 'att1.doc', originalname: 'Attachment 1.doc' }
          ]
        },
        {
          sharedId: 'ent2',
          file: { filename: 'main2.pdf', originalname: 'Main 2.pdf' },
          attachments: [
            { _id: 'att2', filename: 'att2.doc', originalname: 'Attachment 2.doc' }
          ]
        }
      ];
      const res = removeEntityFilenames(entities);
      expect(res).toMatchSnapshot();
    });
  });
});
