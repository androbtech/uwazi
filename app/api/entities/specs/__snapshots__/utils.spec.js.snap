// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entities utils removeEntityFilenames should remove filename from main file and attachments for entity 1`] = `
Array [
  Object {
    "attachments": Array [
      Object {
        "_id": "att1",
        "originalname": "Attachment 1.doc",
      },
    ],
    "file": Object {
      "originalname": "Main.pdf",
    },
    "sharedId": "ent1",
  },
  Object {
    "attachments": Array [
      Object {
        "_id": "att2",
        "originalname": "Attachment 2.doc",
      },
    ],
    "file": Object {
      "originalname": "Main 2.pdf",
    },
    "sharedId": "ent2",
  },
]
`;
